# Laboratory excercise 5

The notebook contains excercise connected to the auditory excerice 6. For any questions feel free to contact assistant: eda.jovicic@fer.hr

In [1]:
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf

# set seed for consistency
np.random.seed(2)

1. Load the cleaned dataset from the first exercise. If you haven't saved the dataset, rerun the exercise and save the final dataset.

In [2]:
#load data
df = pd.read_csv("students_score_cleaned.csv")
df.drop(columns=["Unnamed: 0"], inplace=True)
df

EthnicGroup  ParentEduc  TestPrep  ParentMaritalStatus  PracticeSport  \
0                2           1         1                    1              1   
1                2           5         1                    1              2   
2                1           3         1                    2              2   
3                0           0         1                    1              0   
4                2           5         1                    1              2   
...            ...         ...       ...                  ...            ...   
29112            3           2         1                    2              2   
29113            4           2         1                    2              1   
29114            2           2         0                    1              2   
29115            3           0         0                    1              1   
29116            1           5         1                    1              0   

       IsFirstChild  NrSiblings  TransportMeans  WklyStudyHours  MathScore  \
0                 1         3.0               1               1         71   
1                 1         0.0               2               0         69   
2                 1         4.0               1               1         87   
3                 0         1.0               2               0         45   
4                 1         0.0               1               0         76   
...             ...         ...             ...             ...        ...   
29112             0         2.0               1               0         59   
29113             0         1.0               0               0         58   
29114             0         1.0               0               0         61   
29115             0         3.0               1               0         82   
29116             0         1.0               1               0         64   

       ReadingScore  WritingScore  Gender_female  Gender_male  \
0                71            74            1.0          0.0   
1                90            88            1.0          0.0   
2                93            91            1.0          0.0   
3                56            42            0.0          1.0   
4                78            75            0.0          1.0   
...             ...           ...            ...          ...   
29112            61            65            1.0          0.0   
29113            53            51            0.0          1.0   
29114            70            67            1.0          0.0   
29115            90            93            1.0          0.0   
29116            60            58            0.0          1.0   

       LunchType_free/reduced  LunchType_standard  
0                         0.0                 1.0  
1                         0.0                 1.0  
2                         0.0                 1.0  
3                         1.0                 0.0  
4                         0.0                 1.0  
...                       ...                 ...  
29112                     0.0                 1.0  
29113                     0.0                 1.0  
29114                     1.0                 0.0  
29115                     0.0                 1.0  
29116                     0.0                 1.0  

[29117 rows x 16 columns]

2. Using linear regression, model the MathScore using features: TestPrep, NrSiblings, ParentMaritalStatus and ReadingScore. 

In [3]:
# Declares the model
model = smf.ols(formula='MathScore ~ TestPrep + NrSiblings + ParentMaritalStatus + ReadingScore', data=df)
model.fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              MathScore   R-squared:                       0.669
Model:                            OLS   Adj. R-squared:                  0.669
Method:                 Least Squares   F-statistic:                 1.474e+04
Date:                Tue, 09 Jan 2024   Prob (F-statistic):               0.00
Time:                        23:30:27   Log-Likelihood:            -1.0471e+05
No. Observations:               29117   AIC:                         2.094e+05
Df Residuals:                   29112   BIC:                         2.095e+05
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               6.3552      0.302     21.030      0.000       5.763       6.947
TestPrep                1.1073      0.113      9.817      0.000       0.886       1.328
NrSiblings              0.0154      0.036      0.426      0.670      -0.056       0.086
ParentMaritalStatus    -0.0410      0.076     -0.536      0.592      -0.191       0.109
ReadingScore            0.8570      0.004    239.501      0.000       0.850       0.864
==============================================================================
Omnibus:                      233.653   Durbin-Watson:                   1.987
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              155.726
Skew:                          -0.010   Prob(JB):                     1.53e-34
Kurtosis:                       2.642   Cond. No.                         422.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Questions:

a) Using the summary, write down the formula with coefficients

$MathScore = 1.1073*TestPrep + 0.0154*NrSiblings - 0.0410*ParentMaritalStatus + 0.8570*ReadingScore + 6.3552 $

b) Are there any statisticaly important features?

**Yes, the statistically important features are TestPrep and ReadingScore**

c) Let's interpret the data, if students do test preparation, how does that affect the outcome on the math exam? 

**Test preparation affects the outcome positively (the coefficient is positive).**

d) Are students who practice sport better at math exams or worse? Using linear regression, model the MathScore using only one feature: PracticeSport. Don't include intercept. Using summary, decide which students do the best at math exams - those who never practice sport, sometimes practice sport or those who do it regularly?

In [4]:
# Declare the model
#lr predict MathScore with PracticeSport without intercept
model = smf.ols(formula='MathScore ~ PracticeSport -1', data=df)
model.fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:              MathScore   R-squared (uncentered):                   0.753
Model:                            OLS   Adj. R-squared (uncentered):              0.753
Method:                 Least Squares   F-statistic:                          8.857e+04
Date:                Tue, 09 Jan 2024   Prob (F-statistic):                        0.00
Time:                        23:30:28   Log-Likelihood:                     -1.4397e+05
No. Observations:               29117   AIC:                                  2.879e+05
Df Residuals:                   29116   BIC:                                  2.880e+05
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
PracticeSport    38.3548      0.129    297.600      0.000      38.102      38.607
==============================================================================
Omnibus:                     1585.882   Durbin-Watson:                   1.658
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1060.414
Skew:                           0.352   Prob(JB):                    5.42e-231
Kurtosis:                       2.384   Cond. No.                         1.00
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

**With a high positive coefficient, we can say that practicing sports influences the Math score positively. Having a higher PracticeSport value provides a more significant increase. In our dataset, values for practicing sports are 0=never, 1=regularky, 2=sometimes. This means that students who practice sports sometimes get a higher Maths score.**